src/main/java/org/opensearch/security/auditlog/sink/AbstractInternalOpenSearchSink.java

```diff
@@ -62,6 +88,8 @@ public boolean doStore(final AuditMessage msg, String indexName) {
             return true;
         }
 
+        createIndexIfAbsent(indexName);
+
         try (StoredContext ctx = threadPool.getThreadContext().stashContext()) {
```

Comment:
I think we will need to move this within the try/catch post stashing context - 
1. Moving to try/catch will help handle exceptions during createIndex
2. You might need to stashContext of the thread to allow index to be created for requests that do not have access to create index. (needs testing)